tracer/src/Datadog.Trace/AppSec/Rasp/RaspModule.cs

```diff
@@ -53,8 +53,36 @@ private static void RunWaf(Dictionary<string, object> arguments, Span rootSpan)
         var securityCoordinator = new SecurityCoordinator(Security.Instance, SecurityCoordinator.Context, rootSpan);
         var result = securityCoordinator.RunWaf(arguments, runWithEphemeral: true);
 
+        if (result?.SendStackInfo is not null && Security.Instance.Settings.StackTraceEnabled)
+        {
+            result.SendStackInfo.TryGetValue("stack_id", out var stackIdObject);
+            var stackId = stackIdObject as string;
+
+            if (stackId is null || stackId.Length == 0)
+            {
+                Log.Warning("RASP: A stack was received without Id.");
```

Comment:
Is it possible for this to happen in "normal" circumstances? If so, then you should maybe downgrade this to Debug, to avoid flooding the logs. If not (i.e. only a bug can cause it) then this is fine

Reply:
It should never happen, the id should always be provided by the WAF